db/src/main/java/com/readrops/db/entities/Folder.kt

```diff
@@ -20,4 +23,10 @@ data class Folder(
 ) : Comparable<Folder> {
 
     override fun compareTo(other: Folder): Int = this.name!!.compareTo(other.name!!)
-}
\ No newline at end of file
+}
+
+data class FolderWithFeedAndUnreadCount(
+    @Embedded val folder: Folder?,
+    @Relation(parentColumn = "id", entityColumn = "folder_id")
+    val feeds: List<FeedAndUnreadCount>
+)
```

Comment:
This groups folder and their feeds with unread count. I couldn't get the DB to compute unread counts by folder. The process requires an aggregation function which is destructive unless you use [window functions](https://sqlite.org/windowfunctions.html). Sadly, [this requires SQLite 3.25.0](https://stackoverflow.com/a/55069907), which Android [embeds only until API 30](https://developer.android.com/reference/android/database/sqlite/package-summary). So unread counts will have to be computed from Kotlin until this app cease to support Android 10 and lower.